File: python-pr/52/python/code-quality-readability-maintainability/json_flattener_22.py
# {fact rule=code-quality-readability-maintainability@v1.0 defects=1}
def flatten_json(data, parent_key='', sep='_'):
    items = []
    if isinstance(data, dict):
        for k, v in data.items():
            new_key = f"{parent_key}{sep}{k}" if parent_key else k
            if isinstance(v, dict):
                items.extend(flatten_json(v, new_key, sep).items())
            elif isinstance(v, list):
                for i, item in enumerate(v):
                    items.extend(flatten_json(item, f"{new_key}{sep}{i}", sep).items())
            else:
                items.append((new_key, v))
    return dict(items)
# {/fact}

File: js-pr/56/javascript/code-quality-readability-maintainability/json_flattener_22.js
// {fact rule=code-quality-readability-maintainability@v1.0 defects=1}
function flattenJson(data, parentKey = '', sep = '_') {
    let items = [];
    if (typeof data === 'object' && data !== null) {
        for (let [k, v] of Object.entries(data)) {
            let newKey = parentKey ? `${parentKey}${sep}${k}` : k;
            if (typeof v === 'object' && v !== null && !Array.isArray(v)) {
                items.push(...Object.entries(flattenJson(v, newKey, sep)));
            } else if (Array.isArray(v)) {
                for (let i = 0; i < v.length; i++) {
                    items.push(...Object.entries(flattenJson(v[i], `${newKey}${sep}${i}`, sep)));
                }
            } else {
                items.push([newKey, v]);
            }
        }
    }
    return Object.fromEntries(items);
}
// {/fact}